projects/plugins/boost/app/assets/src/js/pages/settings/elements/Module.svelte

```diff
@@ -13,9 +13,18 @@
 
 	async function handleToggle() {
 		const toggledState = ! isModuleActive;
-		$modulesState[ slug ].active = toggledState;
 		const eventName = toggledState === true ? 'enabled' : 'disabled';
-		dispatch( eventName );
+
+		// Toggle the module.
+		$modulesState[ slug ].active = toggledState;
+
+		// Attach a listener to the modulesStatePending store to dispatch an event after the pending state is resolved.
+		const unsubscribe = modulesStatePending.subscribe( isPending => {
+			if ( ! isPending ) {
+				dispatch( eventName );
+				unsubscribe();
+			}
+		} );
```

Comment:
This code is more complex than it needs to be. It's in a svelte component, so we can just use a `$:` block to react to changes in a stored value - including pending state.

In d5d5518368 I replaced this subscribe-unsubscribe stuff with just a separate block that watches for changes to the store and emits when it's different and not pending.

It also means that if saving the new state of the toggle fails, it won't trigger a dispatch falsely indicating the state changed.

Take a look, see if you're happy with it.